### ISSUE TITLE
Rule is empty when defining root-only CSS variables

### ISSUE THREAD

**Issue by eliperelman** [CSSLint/csslint#538]
When using csslint with variables against the `:root` element, it throws the message of "rule is empty".

``` css
:root {
  --color-alpha: #fff;
}
```

This rule should not be considered empty against the `:root` element that only contains CSS variables as it is considered the de-facto way to create them globally.

**Comment by Mottie** [comment]
+1, also:

* If you add more than one variable, a `Expected RBRACE` error pops up.
* Using a variable in the css then shows this warning `Expected (<color> | inherit) but found 'var(--color-alpha)'.`

**Comment by gabriellecrawford** [comment]
caused by https://github.com/CSSLint/parser-lib/issues/120?

**Comment by Mottie** [comment]
@tophf Has fixed this problem, and many others, in our heavily modified version of [`csslint-worker.js`](https://github.com/openstyles/stylus/blob/master/vendor-overwrites/csslint/csslint-worker.js) in case anyone is interested.

**Comment by dcloutman** [comment]
Still broken.

**Comment by seanw** [comment]
@Mottie @tophf Thanks for suggesting a workaround and your work on this. Can you confirm the license of your modified version? https://github.com/openstyles/stylus/tree/master/vendor-overwrites/csslint

**Comment by tophf** [comment]
We didn't change the license:

>Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

**Comment by seanw** [comment]
@tophf Thanks, the repo is marked as GPL so I wanted to be 100% sure here that you're allowing others to use the modifications under the original non-GPL license.

**Comment by Arcanemagus** [comment]
You could also just use a modern (and maintained) style code linter like [stylelint](https://stylelint.io/) 😉.

**Comment by Mottie** [comment]
Stylelint doesn't validate the css. The authors are deferring the validation to [csstree](https://github.com/csstree/stylelint-validator).

That and csslint is less than 100KB in size while Stylelint (without csstree) is over 3MB in size and does not work with browserify nor rollup &ndash; which is why we ended up creating the [stylelint-bundle repository](https://github.com/openstyles/stylelint-bundle) which removes some of the functionality to get stylelint to bundle and work properly.

**Comment by seanw** [comment]
@Mottie Something that can be important as well is csslint doesn't use eval anywhere which isn't the case for csstree.

**Comment by rtm** [comment]
@seanw From a functionality perspective, one con of csslint is that it doesn't work on modern CSS including custom properties.

**Comment by Ademord** [comment]
What is the solution of this?
